File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/surveys/repository/jpa/inmemory/SurveyRepositoryInMemory.java
package io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.inmemory;

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*;
import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.SurveyRepository;
import java.math.*;
import java.time.*;
import java.util.*;

public class SurveyRepositoryInMemory extends InMemoryJpaRepository<Survey> implements SurveyRepository {
    @Override
    public java.util.Optional<Survey> findByNameAndHospitalId(String name, Long hospitalId) {
        return getEntities().values().stream()
                .filter(e ->
                        isSameValue(name, readField(e, "name")) && isSameValue(hospitalId, readField(e, "hospitalId")))
                .findFirst();
    }
}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/surveys/repository/jpa/inmemory/SurveyRepositoryInMemory.kt
package io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.inmemory

import io.zenwave360.example.clinicaltool.modules.surveys.domain.*
import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.SurveyRepository
import java.math.*
import java.time.*
import java.util.*

class SurveyRepositoryInMemory : InMemoryJpaRepository<Survey, Long>(), SurveyRepository {
    override fun findByNameAndHospitalId(name: String, hospitalId: Long): Survey? {
        return getEntities().values.firstOrNull { e ->
            isSameValue(name, readField(e, "name")) && isSameValue(hospitalId, readField(e, "hospitalId"))
        }
    }
}
